src/test.rs

```diff
@@ -158,6 +158,53 @@ fn cross_process_embedded_senders() {
     assert_eq!(received_person, person);
 }
 
+#[cfg(not(any(feature = "force-inprocess", target_os = "windows", target_os = "android", target_os = "ios")))]
+#[test]
+fn stress_in_process() {
+    enum Message {
+        CloneApi(IpcSender<i32>),
+    }
+
+    impl Serialize for Message {
+        fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error> where S: Serializer {
+            match *self {
+                Message::CloneApi(ref s) => s.serialize(serializer),
+            }
+        }
+    }
+
+    impl<'de> Deserialize<'de> for Message {
+        fn deserialize<D>(deserializer: D) -> Result<Self, D::Error> where D: Deserializer<'de> {
+            Deserialize::deserialize(deserializer).map(Message::CloneApi)
+        }
+    }
+
+
+    let (api, rx) = ipc::channel::<Message>().unwrap();
+
+    {
+        ::std::mem::forget(api.clone());
+    }
```

Comment:
Are you doing this so the sub-thread lingers around rather than panicking when the test is done? That's not OK in my book... Just let the thread exit cleanly when it gets a "channel closed" notification.

In fact I think we should even join the thread explicitly after dropping the channel, just to be sure everything went fine and we don't leave any garbage around.

(Also, I'm not sure what the extra scope is supposed to do here?...)